scrapscript_tests.py

```diff
@@ -4052,44 +4052,46 @@ def test_pretty_print_variant(self) -> None:
 
 
 class ServerCommandTests(unittest.TestCase):
+    import urllib.request
+
     def setUp(self) -> None:
         import threading, time, os, socket, argparse
         from scrapscript import server_command
 
         # Find a random available port
         with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
             s.bind(("127.0.0.1", 0))
-            self.port = s.getsockname()[1]
+            self.host, self.port = s.getsockname()
 
         args = argparse.Namespace(
             directory=os.path.abspath(os.path.join(os.path.dirname(__file__), "examples")),
-            host="127.0.0.1",
+            host=self.host,
             port=self.port,
         )
 
         self.server_thread = threading.Thread(target=server_command, args=(args,))
         self.server_thread.daemon = True
         self.server_thread.start()
 
-        time.sleep(0.1)
+        # Wait for the server to start
+        while True:
+            try:
+                with socket.create_connection((self.host, self.port), timeout=0.1) as s: 
+                    break
+            except Exception:
```

Comment:
Can you make this Exception more specific? Otherwise a typo could cause AttributeError which would cause this to loop indefinitely instead of raising